### ISSUE TITLE
Connection problems

### ISSUE THREAD

**Issue by DavidCGN** [nielsengelen/vbo365-rest#92]
Hello,

im currently struggling to establish a connection between the portal and our Office365-Server.
Im currently looking for some issues with my port configuration. Every firewall is disabled and there is still no connection.
Im using Ubuntu-Server and configurated everything with those documentations you provided.
Can somebody help me troubleshooting that error?
Here the error message:
Error creating resource: [message] fopen(https://***.***.***.***:4443/v5/token): failed to open stream: Connection timed out [file] /var/www/html/vendor/guzzlehttp/guzzle/src/Handler/StreamHandler.php [line] 324.

Thanks in advance!

**Comment by nielsengelen** [comment]
Do you get a ping response from the VBO server when performing it on the Ubuntu server? You could try to do a nmap portscan (nmap -P0 -p 4443 IP) to see if the port is copen.

**Comment by DavidCGN** [comment]
Ping is working from the vbo server to the Ubuntu server. I just checked the port connection via nmap. It seems like only port 80 and 22 are working right now. I'll check that again on the server.

**Comment by DavidCGN** [comment]
Ok, I think I found the problem and fixed it maybe (the restful api wasnt enabled in Windows). 

Now Im running into new errors.
Error creating resource: [message] fopen(): SSL: Handshake timed out [file] /var/www/html/vendor/guzzlehttp/guzzle/src/Handler/StreamHandler.php [line] 324 [message] fopen(): Failed to enable crypto [file] /var/www/html/vendor/guzzlehttp/guzzle/src/Handler/StreamHandler.php [line] 324 [message] fopen(https://******:4443/v5/token): failed to open stream: operation failed [file] /var/www/html/vendor/guzzlehttp/guzzle/src/Handler/StreamHandler.php [line] 324

I can't find a solution for this. Maybe you have some ideas.

Thanks in advance for your help!

**Comment by nielsengelen** [comment]
Are you using a self-signed certificate for the RESTful API service?

**Comment by DavidCGN** [comment]
Yes I am. I changed the value already to verify in the veeam.class.php.

**Comment by DavidCGN** [comment]
Ok, I was troubleshooting again and now I have a new error.  I found out, that I didnt install php curl and that was a part of the problems. 
Now with php curl installed I have this error: 
cURL error 28: Operation timed out after 30001 milliseconds with 0 out of 0 bytes received

I will look into that again.

**Comment by nielsengelen** [comment]
That sounds like the Linux server is unable to reach the VBO server at all. Are you connecting to the IP or hostname from the VBO server?

**Comment by DavidCGN** [comment]
Im currently trying to get the port open on the Linux Server. All Firewalls are still down. Ping is working between those two. A tnc on port 4443 is not coming through. I added the port to the ufw firewall but that didnt solve the problem. Im still on it.

**Comment by nielsengelen** [comment]
Any update on the firewalling issue?

**Comment by DavidCGN** [comment]
Yes, sorry for the late answer. After I checked everything on the hosts I couldnt solve the problem. I made some tests with a windows based installation and had a SSL-Handshake error. The problem was our firewall that forwards the connection. I changed the rules there to allow any sourceport with the destinationport 4443 to come through (before that it was source 4443 to destination 4443).
Now it looks like its working fine.
